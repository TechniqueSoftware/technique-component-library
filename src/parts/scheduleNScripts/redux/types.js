export const key = 'scheduleNScripts'

const FOLLOWUPS = {
  GET: `${key}/FOLLOWUPS/GET`,
  ERROR: `${key}/FOLLOWUPS/ERROR`,
  SUCCESS: `${key}/FOLLOWUPS/SUCCESS`,
  PERMISSION: `${key}/FOLLOWUPS/PERMISSION`,
  SET: `${key}/FOLLOWUPS/SET`,
  INNER_PERMISSIONS: `${key}/FOLLOWUPS/INNER_PERMISSIONS`,
  SET_INNER_PERMISSIONS: `${key}/FOLLOWUPS/SET_INNER_PERMISSIONS`,
  GET_FEATURE_PERMISSIONS: `${key}/FOLLOWUPS/GET_FEATURE_PERMISSIONS`
}

const TOUCHPOINT = {
  OPEN: `${key}/TOUCHPOINT/OPEN`,
  EDIT: `${key}/TOUCHPOINT/EDIT`,
  EDIT_TOUCHPOINT: `${key}/TOUCHPOINT/EDIT_TOUCHPOINT`,
  EDIT_SUCCESS: `${key}/TOUCHPOINT/EDIT_SUCCESS`,
  EDIT_ERROR: `${key}/TOUCHPOINT/EDIT_ERROR`,
  NEW: `${key}/TOUCHPOINT/NEW`,
  NEW_SUCCESS: `${key}/TOUCHPOINT/NEW_SUCCESS`,
  NEW_ERROR: `${key}/TOUCHPOINT/NEW_ERROR`,
  DELETE: `${key}/TOUCHPOINT/DELETE`,
  DELETE_SUCCESS: `${key}/TOUCHPOINT/DELETE_SUCCESS`,
  DELETE_ERROR: `${key}/TOUCHPOINT/DELETE_ERROR`,
  CLOSE: `${key}/TOUCHPOINT/CLOSE`
}

const SCRIPTS = {
  GET: 'scheduleNScripts/SCRIPTS/GET',
  SAVE: 'scheduleNScripts/SCRIPTS/SAVE',
  EDIT: 'scheduleNScripts/SCRIPTS/EDIT'
}

const SEGMENTS = {
  CLOSE: `${key}/SEGMENTS/CLOSE`,
  GET: `${key}/SEGMENTS/GET`,
  SET: `${key}/SEGMENTS/SET`,
  GET_ATTRIBUTES: `${key}/SEGMENTS/GET_ATTRIBUTES`,
  SET_ATTRIBUTES: `${key}/SEGMENTS/SET_ATTRIBUTES`,
  SET_DATA: `${key}/SEGMENTS/SET_DATA`,
  SAVE: `${key}/SEGMENTS/SAVE`,
  DELETE: `${key}/SEGMENTS/DELETE`,
  GET_LOCATIONS: `${key}/SEGMENTS/GET_LOCATIONS`,
  SET_LOCATIONS: `${key}/SEGMENTS/SET_LOCATIONS`,
  GET_LIST: `${key}/SEGMENTS/GET_LIST`,
  SET_LIST: `${key}/SEGMENTS/SET_LIST`,
  GET_COPY_PROCESS: `${key}/SEGMENTS/GET_COPY_PROCESS`,
  SET_COPY_PROCESS: `${key}/SEGMENTS/SET_COPY_PROCESS`,
  SET_ALERT: `${key}/SEGMENTS/SET_ALERT`,
  ERROR: `${key}/SEGMENTS/ERROR`
}

const SNACKBAR = {
  CLOSE: `${key}/SNACKBAR/CLOSE`,
  SET: `${key}/SNACKBAR/SET`
}

export default {
  FOLLOWUPS,
  TOUCHPOINT,
  SCRIPTS,
  SEGMENTS,
  SNACKBAR
}
