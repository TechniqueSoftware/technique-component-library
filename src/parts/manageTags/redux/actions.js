import TYPES from './types'

const DATA_ACTIONS = {
  SET: payload => ({ type: TYPES.DATA.SET, payload }),

  REQUEST_TAGS: payload => ({ type: TYPES.DATA.REQUEST_TAGS, payload }),
  SAVE_TAGS: payload => ({ type: TYPES.DATA.SAVE_TAGS, payload }),
  DELETE_TAGS: payload => ({ type: TYPES.DATA.DELETE_TAGS, payload })
}

export default {
  DATA: DATA_ACTIONS
}
