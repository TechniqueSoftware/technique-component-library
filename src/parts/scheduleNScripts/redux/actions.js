import TYPES from './types'

const FOLLOWUPS_ACTIONS = {
  GET: () => ({ type: TYPES.FOLLOWUPS.GET }),
  ERROR: payload => ({ type: TYPES.FOLLOWUPS.ERROR, payload }),
  SUCCESS: payload => ({ type: TYPES.FOLLOWUPS.SUCCESS, payload }),
  PERMISSION: payload => ({ type: TYPES.FOLLOWUPS.PERMISSION, payload }),
  SET: payload => ({ type: TYPES.FOLLOWUPS.SET, payload }),
  INNER_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.INNER_PERMISSIONS, payload }),
  SET_INNER_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.SET_INNER_PERMISSIONS, payload }),
  GET_FEATURE_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.GET_FEATURE_PERMISSIONS, payload })
}

const TOUCHPOINT_ACTIONS = {
  OPEN: payload => ({ type: TYPES.TOUCHPOINT.OPEN, payload }),
  EDIT: payload => ({ type: TYPES.TOUCHPOINT.EDIT, payload }),
  EDIT_TOUCHPOINT: payload => ({ type: TYPES.TOUCHPOINT.EDIT_TOUCHPOINT, payload }),
  EDIT_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.EDIT_SUCCESS, payload }),
  EDIT_ERROR: payload => ({ type: TYPES.TOUCHPOINT.EDIT_ERROR, payload }),
  NEW: payload => ({ type: TYPES.TOUCHPOINT.NEW, payload }),
  NEW_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.NEW_SUCCESS, payload }),
  NEW_ERROR: payload => ({ type: TYPES.TOUCHPOINT.NEW_ERROR, payload }),
  DELETE: payload => ({ type: TYPES.TOUCHPOINT.DELETE, payload }),
  DELETE_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.DELETE_SUCCESS, payload }),
  DELETE_ERROR: payload => ({ type: TYPES.TOUCHPOINT.DELETE_ERROR, payload }),
  CLOSE: () => ({ type: TYPES.TOUCHPOINT.CLOSE })
}

const SCRIPT_ACTIONS = {
  GET: () => ({ type: TYPES.SCRIPTS.GET }),
  SAVE: payload => ({ type: TYPES.SCRIPTS.SAVE, payload }),
  EDIT: payload => ({ type: TYPES.SCRIPTS.EDIT, payload })
}

const SEGMENTS_ACTIONS = {
  CLOSE: () => ({ type: TYPES.SEGMENTS.CLOSE }),
  GET: payload => ({ type: TYPES.SEGMENTS.GET, payload }),
  SET: payload => ({ type: TYPES.SEGMENTS.SET, payload }),
  GET_ATTRIBUTES: () => ({ type: TYPES.SEGMENTS.GET_ATTRIBUTES }),
  SET_ATTRIBUTES: payload => ({ type: TYPES.SEGMENTS.SET_ATTRIBUTES, payload }),
  SET_DATA: payload => (({ type: TYPES.SEGMENTS.SET_DATA, payload })),
  SAVE: payload => ({ type: TYPES.SEGMENTS.SAVE, payload }),
  DELETE: payload => ({ type: TYPES.SEGMENTS.DELETE, payload }),
  GET_LOCATIONS: () => ({ type: TYPES.SEGMENTS.GET_LOCATIONS }),
  SET_LOCATIONS: payload => ({ type: TYPES.SEGMENTS.SET_LOCATIONS, payload }),
  GET_LIST: payload => ({ type: TYPES.SEGMENTS.GET_LIST, payload }),
  SET_LIST: payload => ({ type: TYPES.SEGMENTS.SET_LIST, payload }),
  GET_COPY_PROCESS: payload => ({ type: TYPES.SEGMENTS.GET_COPY_PROCESS, payload }),
  SET_COPY_PROCESS: payload => ({ type: TYPES.SEGMENTS.SET_COPY_PROCESS, payload }),
  SET_ALERT: payload => ({ type: TYPES.SEGMENTS.SET_ALERT, payload }),
  ERROR: payload => ({ type: TYPES.SEGMENTS.ERROR, payload })
}

const SNACKBAR_ACTIONS = {
  CLOSE: () => ({ type: TYPES.SNACKBAR.CLOSE }),
  SET: payload => ({ type: TYPES.SNACKBAR.SET, payload })
}

export default {
  FOLLOWUPS: FOLLOWUPS_ACTIONS,
  TOUCHPOINT: TOUCHPOINT_ACTIONS,
  SCRIPTS: SCRIPT_ACTIONS,
  SEGMENTS: SEGMENTS_ACTIONS,
  SNACKBAR: SNACKBAR_ACTIONS
}
