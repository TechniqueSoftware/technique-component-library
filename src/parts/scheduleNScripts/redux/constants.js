export const TP_OPTIONS = {
  MENU: 'menu',
  NEW: 'new',
  EDIT: 'edit'
}

export const GROUP_NAMES = {
  PROSPECT: 'Prospect',
  ACTIVE_MEMBER: 'Active Member',
  INACTIVE_MEMBER: 'Inactive Member',
  PT_PROSPECT: 'PT Prospect',
  PT_MEMBER: 'PT Member',
  PT_OPT_OUT: 'PT Opt Out',
  PERSONAL_TRAINING: 'Personal Training'
}
