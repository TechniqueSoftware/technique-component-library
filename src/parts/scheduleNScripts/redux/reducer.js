import { produce } from 'immer'
import { TP_OPTIONS } from './constants'
import TYPES from './types'

export const initialState = {
  loading: false,
  error: '',
  snackbar: '',
  followups: [],
  canEditScripts: false,
  canEditSchedules: false,
  touchpointCrud: {
    action: '',
    data: {},
    loading: false
  },
  script: {},
  segments: [],
  segmentModalData: {
    isOpen: false,
    title: '',
    error: ''
  },
  segmentAttributes: {
    marketing: [],
    membership: []
  },
  innerPermissions: {
    automaticFollowUpAction: false,
    followUpSegmentsAction: false
  },
  segmentLocations: [],
  segmentList: [],
  copyInProcess: {},
  popupAlert: ''
}

const reducer = (state = initialState, { type, payload }) => produce(state, draft => {
  switch (type) {
    case TYPES.FOLLOWUPS.GET:
      draft.loading = true
      draft.followups = []
      draft.error = ''
      break

    case TYPES.FOLLOWUPS.ERROR:
      draft.loading = false
      draft.error = payload
      draft.followups = []
      break

    case TYPES.FOLLOWUPS.SUCCESS:
      draft.loading = false
      draft.error = ''
      draft.followups = payload.followups
      draft.canEditSchedules = payload.canEditSchedules
      draft.canEditScripts = payload.canEditScripts
      break

    case TYPES.FOLLOWUPS.PERMISSION:
      draft.canEditSchedules = payload.canEditSchedules
      draft.canEditScripts = payload.canEditScripts
      break

    case TYPES.FOLLOWUPS.SET:
      draft.followups = payload
      break

    case TYPES.FOLLOWUPS.SET_INNER_PERMISSIONS:
      draft.innerPermissions = payload
      break

    case TYPES.TOUCHPOINT.OPEN:
      draft.touchpointCrud = {
        action: payload.action,
        data: payload.data
      }
      break

    case TYPES.TOUCHPOINT.CLOSE:
      draft.touchpointCrud = { action: '', data: {}, loading: false }
      break

    case TYPES.TOUCHPOINT.NEW:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.NEW_SUCCESS:
      draft.followups = payload
      draft.touchpointCrud.loading = false
      draft.snackbar = 'NEW_SUCCESS'
      break

    case TYPES.TOUCHPOINT.NEW_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.TOUCHPOINT.DELETE:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.DELETE_SUCCESS:
      draft.followups = payload
      draft.touchpointCrud.loading = false
      draft.snackbar = 'DELETE_SUCCESS'
      break

    case TYPES.TOUCHPOINT.DELETE_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.TOUCHPOINT.EDIT:
      draft.touchpointCrud.action = TP_OPTIONS.EDIT
      break

    case TYPES.TOUCHPOINT.EDIT_TOUCHPOINT:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.EDIT_SUCCESS:
      draft.followups = payload.followups
      draft.touchpointCrud.loading = false
      draft.snackbar = payload.snackbar
      break

    case TYPES.TOUCHPOINT.EDIT_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.SNACKBAR.CLOSE:
      draft.snackbar = ''
      break

    case TYPES.SNACKBAR.SET:
      draft.snackbar = payload
      break

    case TYPES.SCRIPTS.EDIT:
      draft.touchpointCrud = {
        action: payload.action,
        data: payload.data
      }
      break

    case TYPES.SCRIPTS.SAVE:
      draft.script = {
        data: payload.data
      }
      break

    case TYPES.SEGMENTS.SET:
      draft.segments = payload
      break

    case TYPES.SEGMENTS.SET_DATA:
      draft.segmentModalData.title = payload.title
      draft.segmentModalData.isOpen = true
      draft.segmentModalData.followUpType = payload.followUpType
      draft.segmentModalData.followUpStatus = payload.followUpStatus
      break

    case TYPES.SEGMENTS.SET_ATTRIBUTES:
      draft.segmentAttributes = payload
      break

    case TYPES.SEGMENTS.CLOSE:
      draft.segmentModalData.isOpen = false
      draft.segmentModalData.error = ''
      break

    case TYPES.SEGMENTS.ERROR:
      draft.segmentModalData.error = payload
      break

    case TYPES.SEGMENTS.SET_LOCATIONS:
      draft.segmentLocations = payload
      break

    case TYPES.SEGMENTS.SET_LIST:
      draft.segmentList = payload
      break

    case TYPES.SEGMENTS.SET_ALERT:
      draft.popupAlert = payload
      break

    case TYPES.SEGMENTS.SET_COPY_PROCESS:
      draft.copyInProcess = payload
      break
  }
})

export default reducer
