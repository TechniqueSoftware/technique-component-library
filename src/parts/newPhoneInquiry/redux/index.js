import { createManager } from '@redux/injector'

import reducer from './reducer'
import { key } from './types'
import saga from './sagas'

export { default as REDUCER } from './reducer'
export { default as ACTIONS } from './actions'
export { default as SELECTORS } from './selectors'
export { key, default as TYPES } from './types'

export const Manager = createManager({
  name: 'NewPhoneInquiryManager',
  key,
  reducer,
  saga
})
