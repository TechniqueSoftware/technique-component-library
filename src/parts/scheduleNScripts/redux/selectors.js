import { createSelector } from 'reselect'
import { get } from 'lodash'

import { key } from './types'
import { initialState } from './reducer'

const reducer = state => get(state, key, initialState)

const getLoading = createSelector(
  reducer, state => get(state, 'loading', false)
)

const getError = createSelector(
  reducer, state => get(state, 'error', '')
)

const getSnackbar = createSelector(
  reducer, state => get(state, 'snackbar', '')
)

const getFollowups = createSelector(
  reducer, state => get(state, 'followups', [])
)

const getTouchpointCrud = createSelector(
  reducer, state => get(state, 'touchpointCrud', {})
)

const getSegments = createSelector(
  reducer, state => get(state, 'segments', [])
)

const getSegmentAttributes = createSelector(
  reducer, state => get(state, 'segmentAttributes', { marketing: [], membership: [] })
)

const getSegmentModal = createSelector(
  reducer, state => get(state, 'segmentModalData', { isOpen: false, title: '', error: '' })
)

const getCanEditScripts = createSelector(
  reducer, state => get(state, 'canEditScripts', false)
)

const getCanEditSchedules = createSelector(
  reducer, state => get(state, 'canEditSchedules', false)
)

const getInnerPermissions = createSelector(
  reducer, state => get(state, 'innerPermissions', {})
)

const getSegmentLocations = createSelector(
  reducer, state => get(state, 'segmentLocations', [])
)

const getSegmentList = createSelector(
  reducer, state => get(state, 'segmentList', [])
)

const getCopySegmentProcess = createSelector(
  reducer, state => get(state, 'copyInProcess', {})
)

const getPopupAlert = createSelector(
  reducer, state => get(state, 'popupAlert', '')
)

const getStatus = createSelector(
  [
    reducer,
    (_, { followUpType, followUpStatus }) => ({ followUpType, followUpStatus })
  ],
  (state, { followUpType, followUpStatus }) => {
    const followups = get(state, 'followups', [])

    const status = followups.reduce((acc, curr) => {
      if (curr.type !== followUpType) return acc
      return curr.statuses.find(s => s.id === `${followUpStatus}`) || acc
    }, null)
    return status
  }
)

export default {
  LOADING: getLoading,
  ERROR: getError,
  FOLLOWUPS: getFollowups,
  TOUCHPOINT_CRUD: getTouchpointCrud,
  CAN_EDIT_SCRIPTS: getCanEditScripts,
  CAN_EDIT_SCHEDULES: getCanEditSchedules,
  STATUS: getStatus,
  SEGMENTS: getSegments,
  SEGMENT_MODAL_DATA: getSegmentModal,
  SEGMENT_ATTRIBUTES: getSegmentAttributes,
  SNACKBAR: getSnackbar,
  INNER_PERMISSIONS: getInnerPermissions,
  SEGMENT_LOCATIONS: getSegmentLocations,
  SEGMENT_LIST: getSegmentList,
  SEGMENT_COPY_PROCESS: getCopySegmentProcess,
  POPUP_ALERT: getPopupAlert
}
