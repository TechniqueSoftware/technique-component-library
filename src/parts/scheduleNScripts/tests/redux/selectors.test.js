import { SELECTORS } from '../../redux'

import { state } from './mock'

describe('SELECTORS', () => {
  it('loading', () => {
    expect(SELECTORS.LOADING(state)).toMatchSnapshot()
  })

  it('error', () => {
    expect(SELECTORS.ERROR(state)).toMatchSnapshot()
  })

  it('followups', () => {
    expect(SELECTORS.FOLLOWUPS(state)).toMatchSnapshot()
  })

  it('touchpointCrud', () => {
    expect(SELECTORS.TOUCHPOINT_CRUD(state)).toMatchSnapshot()
  })

  it('canEditScripts', () => {
    expect(SELECTORS.CAN_EDIT_SCRIPTS(state)).toMatchSnapshot()
  })

  it('canEditSchedules', () => {
    expect(SELECTORS.CAN_EDIT_SCHEDULES(state)).toMatchSnapshot()
  })

  it('getStatus', () => {
    expect(SELECTORS.STATUS(state, { followUpType: 'followup', followUpStatus: '1' })).toMatchSnapshot()
  })

  it('segments', () => {
    expect(SELECTORS.SEGMENTS(state)).toMatchSnapshot()
  })

  it('segmentAttributes', () => {
    expect(SELECTORS.SEGMENT_ATTRIBUTES(state)).toMatchSnapshot()
  })
})
