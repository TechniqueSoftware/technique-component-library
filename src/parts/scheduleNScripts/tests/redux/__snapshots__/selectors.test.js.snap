// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SELECTORS canEditSchedules 1`] = `false`;

exports[`SELECTORS canEditScripts 1`] = `false`;

exports[`SELECTORS error 1`] = `"there is an error here"`;

exports[`SELECTORS followups 1`] = `
Array [
  Object {
    "id": "0",
    "name": "Prospecting",
    "statuses": Array [
      Object {
        "customClubLabel": null,
        "displayName": "Active Guest",
        "followUpTab": "0",
        "groupName": "Prospect",
        "hidden": "false",
        "id": "4",
        "name": "Active Guest",
        "schedules": Array [
          Object {
            "autoEnabled": false,
            "canEditSchedules": true,
            "dayNum": 3,
            "daysTillNext": 3,
            "followUpSeqNum": 1,
            "followUpStatus": 4,
            "followUpType": 1,
            "locationId": 4585,
            "scripts": Array [
              Object {
                "actionType": 1,
                "clubId": 659,
                "followUpSeqNum": 10,
                "followUpStatus": 5,
                "followUpType": 1,
                "script": "<p>Hi {{first-name}}! This is {{employee-first}} at {{club-name}}.</p>",
              },
            ],
          },
          Object {
            "autoEnabled": false,
            "canEditSchedules": true,
            "dayNum": 5,
            "daysTillNext": 2,
            "followUpSeqNum": 2,
            "followUpStatus": 4,
            "followUpType": 1,
            "locationId": 4585,
          },
        ],
      },
      Object {
        "customClubLabel": null,
        "displayName": "Expired Guest",
        "followUpTab": "0",
        "groupName": "Prospect",
        "hidden": "false",
        "id": "5",
        "name": "Expired Guest",
        "schedules": Array [
          Object {
            "autoEnabled": false,
            "canEditSchedules": true,
            "dayNum": 0,
            "daysTillNext": 0,
            "followUpSeqNum": 1,
            "followUpStatus": 5,
            "followUpType": 1,
            "locationId": 4585,
          },
        ],
      },
    ],
    "type": 1,
  },
]
`;

exports[`SELECTORS getStatus 1`] = `null`;

exports[`SELECTORS loading 1`] = `true`;

exports[`SELECTORS segmentAttributes 1`] = `
Object {
  "marketing": Array [],
  "membership": Array [],
}
`;

exports[`SELECTORS segments 1`] = `Array []`;

exports[`SELECTORS touchpointCrud 1`] = `Object {}`;
